### ISSUE TITLE
Gun cotton

### ISSUE THREAD

**Issue by madamimadam** [CleverRaven/Cataclysm-DDA#27132]
**Is your feature request related to a problem? Please describe.**  
I have about five gallons of nitric acid and twenty pounds of cotton from my Lab challenge start. I have a basic chemistry text and an advanced chemistry text. And one of the first things you do for fun in chem 101 is making gun cotton. (That or playing with sodium and phosphorous.)

I mean, hell, we're crafting diazepam, but we can't craft this:
https://www.instructables.com/id/Nitrocellulose/

And hell, making nitric acid is easy too (and there's already a formula for that). And you can use paper for nitrocellulose if you don't want to use cotton. Or wood. Or anything else with cellulose in it, in theory, although I don't know how well different things would work.

Making modern gunpowder is a damn tricky operation, as has been noted in discussions over and over again. But making gun cotton, the first smokeless powder, is utterly straightforward. The above recipe is not suitable for use in guns, because it explodes too fast and demolishes them, but it makes a dandy explosive.

And then if you want to make your gun cotton into something that could be packed into a shell and used in modern weapons, well, I'll let someone else take it from there:

"In 1884, the French physicist Paul Vieille invented a smokeless powder that could be used in rifles. He controlled the burn rate by dissolving some of the nitrocellulose in a mixture of ether and alcohol. The remaining undissolved fibers mixed with the dissolved compound and formed a pasty jelly that could be rolled into sheets and cut, or extruded and cut like noodles."

We have ether and alcohol. And can craft either one.

**Describe the solution you'd like**
1) Gun cotton should absolutely be craftable. In its simple form it can't really be used in modern firearms, so it could be just used as a substitute for black powder, and enough of it could also be crafted into bombs. This won't affect balance: it's more powerful than dynamite, but nitric acid is harder to find than ammonia. And anyway, you could just make it the same power as dynamite, if that part is a concern.

2) If making modern gunpowder is out solely due to balance reasons, then so be it, and we stop there. But if the reason it's out is because we're trying to be realistic, then crafting gun cotton that can be used in modern rifles out of the above-craftable gun cotton is trivial. It is definitely of rougher quality than modern gunpowder, and realistically would make guns jam or break more often, but it would work, and would not foul modern barrels or chambers like black powder would.

**Comment by tenmillimaster** [comment]
It's one thing to make guncotton, yes. But good smokeless powder requires extremely good control of particles, while observing safety practices. Too big and it burns too slow, possibly not fast enough to cycle an action. Too small and the pressure spike at the muzzle could be catastrophic. I'd think that part of the process of making your own smokeless gunpowder would end up consuming a few guns. And fingers.

**Comment by madamimadam** [comment]
We know what happens when you try to make meth and fail: you explode, catch on fire, and die, often but not always in that order. And we also know what happens when you try to make benzodiazepines and get the formula off a little bit: there's no way to know beforehand, and then you take them, start hallucinating, and end up with severe brain damage. Or you die. This is not to mention what happens when the cv joint you made turns out to not QUITE be good enough, at 80 mph.

Crafting in the game is always simplified past problems like that, except, expressly, for gun powder. Gun cotton would be much simpler to experiment with than benzodiazepines (how long and how many lab rats does it take to tell that this formula gave them brain damage? Are you a rat behaviorolotist too?) or even ether. And certainly simpler than testing CV joints that you make at high speeds without getting yourself killed, which would require an entire machine shop with some quite specialized equipment.

And that's suspending disbelief on crafting something like a mutagen, which should certainly require the most rigorous possible lab and cleanroom standards. I always wondered why mutagens were craftable: I feel like they should be rare, sought-after, and really hard to find. Not as rare as artifacts, maybe, but similarly prized. But that's an entirely separate discussion.

If this were real life, I would certainly say that one would have to exercise a lot of caution when experimenting with particle sizes, although from what I understand Vieille's version would be easier to deal with because the consistency of the product tells you a lot about exactly how it will burn. I think that one could get a workable smokeless powder from it without losing any fingers, although I would expect to lose a certain amount of guns, or at any rate gun-mockups.

In any case, none of that addresses point 1: explosives simply don't have any of those problems. Small particles, big particles, a mix, they'll all go boom in the end.

**Comment by madamimadam** [comment]
As I said: if the reason for gunpowder not being craftable is just game balance, then that's fine with me, and gun cotton gunpowder shouldn't be craftable. I just think that, at the very least, that should be expressly stated, and that nitrocellulose explosives should be included.

**Comment by JacobKessler** [comment]
Have you checked the math on guncotton and wrecking guns? I ask because 'burning fast' equates to high pressure, and modern guns can handle pressures that would indeed destroy 1884-vintage guns, and indeed for modern rifle ammunition rely on it. It is also true (as you said) that mixing guncotton with some stabilizers is an entirely doable thing, though it might lead to increased fouling of the gun.

I definitely agree that it would probably only be suitable for handloading ammunition, but it does seem like a viable path to handloaded ammunition.

**Comment by kevingranade** [comment]
Please provide sources on the dead simple guncotton recipe (e.g. from the instructables site) being sufficient to act as a replacement for black powder or in a bomb.

Most of your arguments are invalid, you're basically arguing that any lapse in realism justifies any other lapse in realism.

**Comment by vegarbg** [comment]
> you're basically arguing that any lapse in realism justifies any other lapse in realism.

Well, yes. If some lapses in realism are OK "just because", then that's good enough for any of them. Instead of getting stuck on this point, we should instead evaluate "does this change make the game more fun / interesting / challenging?" Personally I think it would be interesting to add more historical chemistry to the game, but I agree that game balance should take precedence.

Here's some links to sections from _A Practical Treatise On The Fabrication Of Matches, Gun Cotton, Colored Fires And Fulminating Powders_:
* [Gun cotton recipe](https://chestofbooks.com/science/chemistry/Matches-Colored-Fires/Chapter-XXII-Fabrication-Of-Gun-Cotton-Nitric-Paper-Pyroxam-Properties.html) (starting from the paragraph "The preparation of nitric cotton is very simple ...")
* [Report of testing that gun cotton](https://chestofbooks.com/science/chemistry/Matches-Colored-Fires/Gun-Cotton-Physical-Report.html), including for artillery where it was found to have many advantages over black powder.

**Comment by tenmillimaster** [comment]
PDF warnings: These documents might be of interest to those in the thread. 

[Conversion of Nitrocellulose to Smokeless Powder](https://apps.dtic.mil/dtic/tr/fulltext/u2/b803012.pdf)

[THERMOCHEMICAL AND BURNING RATE PROPERTIES OF DETERRED US SMALL ARMS PROPELLANTS](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.863.1601&rep=rep1&type=pdf)

**Comment by vegarbg** [comment]
From _Conversion of Nitrocellulose to Smokeless Powder_
>[The air bubbles] are eliminated by further processing through a macaroni press ...

How wonderful, a non-food use for the pasta extruder :smile:

**Comment by tyrael93** [comment]
Sources checked, realism is a go, looks like everything's in order! Craftable smokeless powder doesn't need to be gated behind a mod anymore, can't wait for the PR.

**Comment by vegarbg** [comment]
How about MythBusters ep 40 "Confederate Rocket" where Kari makes gun cotton in the shop?

**Comment by vegarbg** [comment]
Here's a first draft: [GunCotton](https://github.com/vegarbg/GunCotton)

**Comment by vegarbg** [comment]
Added to the CDDA Launcher repo, feel free to check it out via that.

**Comment by kevingranade** [comment]
Those are some great sources by the way, it's just that reading them reinforces my impression that modern gunpowder manufacture is an extremely technical industrial process that is extremely impractical for a single survivor to replicate.

**Comment by tenmillimaster** [comment]
I'd like to see this  get shelved until the survivor Basecamp faction can setup their own little factory for churning out proper single and double base propellants. Making good powder would probably be best done with a quality control lab.

**Comment by JacobKessler** [comment]
> Those are some great sources by the way, it's just that reading them reinforces my impression that modern gunpowder manufacture is an extremely technical industrial process that is extremely impractical for a single survivor to replicate.

It actually looks like the recipe, assuming you start with a cotton rag rather than cotton balls, is in fact relatively simple: soak the cotton for 15-20 minutes, press it between glass, wash it several times including once with a basic solution, and allow to dry for a few days. Sounds like a recipe that takes ~30 minutes per rag with a good batch time, and requires nitric acid, sulfuric acid, a rag, a small amount of lye, and a lot of clean water. Tools would be a chemistry set, a container, and two sheets of glass.

As a gunpowder replacement, it seems like it would actually have the opposite problem as black powder (which, given the publication date, is probably what it's being compared to in the treatise) - it's a high explosive, and thus would be at risk of destroying a gun (and the user's face) if too much is used, and it doesn't seem like we have a good way of simulating that danger right now. Maybe a flag that indicates that a jam should actually be an explosion, coupled with the handloading misfire penalty? It seems like it should be completely applicable as-is as a filler for explosives like grenades, though.

**Comment by kevingranade** [comment]
> It actually looks like the recipe, assuming you start with a cotton rag rather than cotton balls, is in fact relatively simple

Thats the guncotton recipe, and as I said
> At the end of that process, you have a suitable replacement for black powder.

What I am still confident in leaving out-of-bounds is modern "gunpowder".  Practically speaking, what is meant by this is, "a substance you can use to reload a cartridge for a modern firearm and have it work properly".  

> if too much is used

It's not a matter of how much you use, the problem is there is no overlap between "enough to be useful" and "little enough to not blow up your gun".

Please, before you comment make sure you know which substance(s) people are talking about,  and which substance(s) you are talking about.

**Comment by madamimadam** [comment]
And what you, Kevin, seem to be arguing is that if you can't use it as a replacement for modern gunpowder, it shouldn't exist at all. That's silly. You can make a matchhead bomb. You can make dynamite, which is a rather touchy task and a good way to have your eyebrows embedded in the wall behind you if you make the slightest mistake. Making gun cotton (that isn't that touchy) is very simple, to a quality that can be used to make a bomb. So what's your problem with gun cotton bombs?

As for the rest, I suspect that, if I made some gun cotton in my basement, loaded a hundred shells with it, and then fired them all off with no problems, you would still argue that it was not possible to do. But honestly, I don't care. I was mostly just opening this for the bombs, not for the guns.

**Comment by madamimadam** [comment]
Incidentally, here is the easiest current way to make diazepam:
https://docslide.net/documents/new-synthesis-of-diazepam.html
"A new synthesis of diazepam has been devised. It depends critically on three new findings: (1) 3,4-Dihydro-1H-1,4-benzodazepine-2,5-diones, hitherto difficultly accessible, can be made easily and in high yield directly from isotonic anhydrides and glycine. Intermediate in this preparation are o-aminohippuric acids, which can be isolated as their difficultly soluble potassium salts or can be cyclized without isolation to the benzo-1,4-diazepine-2,5-diones. To achieve these yields it is only necesary to add 1 equiv of a weak bace (Na2CO3 or triethylamine) to convert the glycine into its anion in which the amino group is present as the primary amine rather than as its ammonium ion. The optimum procedure makes use of triethylamine as the base and proceeds to 1 in 95% yield without the isolation of the intermediate."

As it happens, I understand most of that. (Although who says 'difficultly soluble'? Jeez.) Do you know what it takes to make triethylamine? How about the monomers that you need to start with? Glycine doesn't look too bad, but I don't see anything right off that lends itself to the equipment a survivor has. And this is just one of the three breakthroughs that make it 'easy' to produce in quantity.

The survivor makes it out of aspirin, salt, and 'oxidizer powder'. I mean, come ON.

**Comment by madamimadam** [comment]
Oh, actually, here's a good one for glycine. Easy and straightforward.
http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC1215768&blobtype=pdf

**Comment by kevingranade** [comment]
> And what you, Kevin, seem to be arguing is that if you can't use it as a replacement for modern gunpowder, it shouldn't exist at all.

Reviewing my comments, it does seem like that, sorry for the lack of clarity.
You've convinced me that:
Guncotton seems feasible to manufacture.
Guncotton manufactured this way seems usable for bomb manufacture.
Specially-prepared guncotton seems viable for crafting a black powder substitute.

> As for the rest, I suspect that, if I made some gun cotton in my basement, loaded a hundred shells with it, and then fired them all off with no problems, you would still argue that it was not possible to do.

I never said it wasn't possible, I said it wasn't practical.

> The survivor makes it out of aspirin, salt, and 'oxidizer powder'. I mean, come ON.

Yea I want to fix that kind of thing too.

**Comment by kevingranade** [comment]
BTW, if you do add a guncotton recipe, it seems it has a *SIGNIFICANT* processing time after the initial reaction, see https://github.com/CleverRaven/Cataclysm-DDA/pull/17140#issuecomment-225387904
Tl;dr ``The washing and drying times of the nitrocellulose were both extended to 48 hours and repeated eight times over.``
